nautobot/core/views/mixins.py

```diff
@@ -0,0 +1,804 @@
+        if not table.rows:
+            messages.warning(
+                request,
+                f"No {self.queryset.model._meta.verbose_name_plural} were selected for deletion.",
+            )
+            return redirect(self.get_return_url(request))
+
+        data.update({"table": table})
+        return Response(data)
+
+
+class ObjectBulkCreateViewMixin(NautobotViewSetMixin, BulkCreateModelMixin):
+    bulk_create_form_class = None
+    bulk_create_widget_attrs = {}
+
+    def _process_bulk_create_form(self, form):
+        # Iterate through CSV data and bind each row to a new model form instance.
+        new_objs = []
+        request = self.request
+        with transaction.atomic():
+            if request.FILES:
+                field_name = "csv_file"
+            else:
+                field_name = "csv_data"
+            headers, records = form.cleaned_data[field_name]
+            for row, data in enumerate(records, start=1):
+                obj_form = self.bulk_create_form_class(data, headers=headers)
+                restrict_form_fields(obj_form, request.user)
+
+                if obj_form.is_valid():
+                    obj = self.form_save(obj_form)
+                    new_objs.append(obj)
+                else:
+                    for field, err in obj_form.errors.items():
+                        form.add_error("csv_data", f"Row {row} {field}: {err[0]}")
+                    raise ValidationError("")
+
+            # Enforce object-level permissions
+            if self.queryset.filter(pk__in=[obj.pk for obj in new_objs]).count() != len(new_objs):
+                raise ObjectDoesNotExist
+
+        # Compile a table containing the imported objects
+        obj_table = self.table_class(new_objs)
+        if new_objs:
+            msg = f"Imported {len(new_objs)} {new_objs[0]._meta.verbose_name_plural}"
+            self.logger.info(msg)
+            messages.success(request, msg)
+        return obj_table
+
+    def bulk_create(self, request):
+        context = {}
+        if request.method == "POST":
+            return self.perform_bulk_create(request)
+        return Response(context)
+
+    def perform_bulk_create(self, request):
+        form_class = self.get_form_class()
+        form = form_class(request.POST)
+        if form.is_valid():
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
+
+
+class ObjectBulkUpdateViewMixin(NautobotViewSetMixin, BulkUpdateModelMixin):
+    filterset_class = None
+    bulk_update_form_class = None
+
+    def _process_bulk_update_form(self, form):
+        request = self.request
+        model = self.model
+        form_custom_fields = getattr(form, "custom_fields", [])
+        form_relationships = getattr(form, "relationships", [])
+        standard_fields = [
+            field
+            for field in form.fields
+            if field not in form_custom_fields + form_relationships + ["pk"] + ["object_note"]
+        ]
```

Comment:
It's not clear what is really happening here? And why are we explicitly adding `["object_note"]` to the end here? A comment on why this is necessary would be helpful!

Reply:
I think this is included in the [PR](https://github.com/nautobot/nautobot/pull/2152#discussion_r942584352) that added Bulk Edit support for Notes Models.

Reply:
See suggested comment wording above.